### PR TITLE
perf(fetcher): backoff retry in get transactions

### PR DESCRIPTION
**Pull request by kien6034**
# PR Description 

This PR updates the transaction fetching mechanism to use an exponential backoff strategy when retries are necessary. Previously, the system would attempt a maximum of five retries with no delay between them. With the new changes, the delay between retries increases exponentially.

### CODE REVIEW

**Comment by kien6034** [comment]
@raphjaph this is my first attempt contributing to the repo. It would be great if you have a look at this PR. Thank you

File: src/index/fetcher.rs [hunk 1]
```diff
@@ -65,26 +65,29 @@ impl Fetcher {
...
-          if retries >= 5 {
-            return Err(anyhow!(
-              "failed to fetch raw transactions after 5 retries: {}",
-              error
-            ));
-          }
-
-          log::info!("failed to fetch raw transactions, retrying: {}", error);
-
-          tokio::time::sleep(Duration::from_millis(100 * u64::pow(2, retries))).await;
```
Comment (raphjaph) [inline src/index/fetcher.rs:82]: We are already doing exponential backoff.

**Review by raphjaph — COMMENTED** [review]
I don't see what this PR changes. We were already doing exponential back-off.

**Comment by kien6034** [comment]
oh my mistake. Lets me just close this PR for now

**Pull request closed without merging**